therapy/etl/drugbank.py

```diff
@@ -67,295 +31,79 @@ def perform_etl(self) -> List[str]:
+
+        # download file
+        r = requests.get(url)
+        if r.status_code == 200:
+            zip_file = zipfile.ZipFile(BytesIO(r.content))
+        else:
+            logger.error("DrugBank download failed with status code:"
+                         f" {r.status_code}")
+            raise requests.HTTPError(r.status_code)
+
+        # unpack file
+        temp_dir = self._src_data_dir / 'temp_drugbank'
+        zip_file.extractall(temp_dir)
+        temp_file = temp_dir / 'drugbank vocabulary.csv'
+        csv_file = self._src_data_dir / f'drugbank_{version}.csv'
+        shutil.move(temp_file, csv_file)
+        shutil.rmtree(temp_dir)
+        logger.info("DrugBank source data download complete.")
 
     def _load_meta(self):
         """Add DrugBank metadata."""
-        meta = SourceMeta(data_license='CC BY-NC 4.0',
-                          data_license_url='https://creativecommons.org/licenses/by-nc/4.0/legalcode',  # noqa: E501
-                          version=self._version,
-                          data_url=self._data_url,
-                          rdp_url='http://reusabledata.org/drugbank.html',
-                          data_license_attributes={
-                              'non_commercial': True,
-                              'share_alike': False,
-                              'attribution': True
-                          })
-        params = dict(meta)
-        params['src_name'] = SourceName.DRUGBANK.value
-        self._database.metadata.put_item(Item=params)
+        meta = {
+            'data_license': 'CC0 1.0',
+            'data_license_url': 'https://creativecommons.org/publicdomain/zero/1.0/',  # noqa: E501
+            'version': self._version,
+            'data_url': 'https://go.drugbank.com/releases/latest#open-data',
+            'rdp_url': 'http://reusabledata.org/drugbank.html',
+            'data_license_attributes': {
+                'non_commercial': False,
+                'share_alike': False,
+                'attribution': False,
+            },
+        }
+        assert SourceMeta(**meta)
+        meta['src_name'] = SourceName.DRUGBANK.value
+        self.database.metadata.put_item(Item=meta)
+
+    def _transform_data(self):
+        """Transform the DrugBank source."""
+        with open(self._src_file, 'r') as file:
+            reader = csv.reader(file)
+            next(reader)  # skip header
+            for row in reader:
+                params = {
+                    'concept_id': f'{NamespacePrefix.DRUGBANK.value}:{row[0]}',
+                }
+                label = row[2]
+                if label:
+                    params['label'] = label
+                aliases = [
+                    a for a in row[1].split(' | ') + row[5].split(' | ') if a
+                ]
+                if aliases:
+                    params['aliases'] = aliases
+                cas_ref = row[3]
+                if cas_ref:
+                    params['other_identifiers'] = [
```

Comment:
Should we be adding UNII and Standard InChI Key to `other_identifiers`?

Reply:
I assumed Alex had a reason to not include them in the first pass -- I'll ask him tomorrow